File: 3626-smallest-divisible-digit-product-i/smallest-divisible-digit-product-i.java
class Solution {
    public int smallestNumber(int n, int t) {

        while (true) {

            int product = 1;
            int num = n;

            while (num > 0) {
                product *= (num % 10);
                num /= 10;
            }

            if (product % t == 0)
                return n;

            n++;
        }
    }
}

File: 3626-smallest-divisible-digit-product-i/smallest-divisible-digit-product-i.py
class Solution:
    def smallestNumber(self, n: int, t: int) -> int:

        while True:

            product = 1
            num = n

            while num > 0:
                product *= num % 10
                num //= 10

            if product % t == 0:
                return n

            n += 1